pennylane/__init__.py

```diff
@@ -56,10 +57,29 @@
 default_config = Configuration("config.toml")
 
 
+def _get_device_entrypoints():
+    """Returns a dictionary mapping the device short name to the
+    loadable entrypoint"""
+    return {entry.name: entry for entry in pkg_resources.iter_entry_points("pennylane.plugins")}
+
+
+def refresh_devices():
+    """Scan installed PennyLane plugins to refresh the device list."""
+
+    # This function does not return anything; instead, it has a side effect
+    # which is to update the global plugin_devices variable.
+
+    # We wish to retain the behaviour of a global plugin_devices dictionary,
+    # as importing pkg_resources can be a very slow operation on systems
+    # with a large number of installed packages.
+    global plugin_devices  # pylint:disable=global-statement
```

Comment:
While global variables should in general be avoided unless necessary, here we need to store globally the loaded plugin devices, to avoid needlessly refreshing when we don't need to. See https://github.com/PennyLaneAI/pennylane/pull/907/files#r525003135 for more context.

Note: we could re-write the device loader to be a _class_ instead, which would allow us to track state without module variables. However, this is likely overkill - `global` variables are not purely global in Python, but remain namespaced.

Reply:
Yep! The `global` statement in Python is a bit of a misnomer. It doesn't really create a true global variable, it just raises the scope of the variable to the module level.